grakn-module-sdk/src/main/java/ai/grakn/graknmodule/http/HttpRequest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.graknmodule.http;
+
+import java.util.Map;
+
+/**
+ * A class representing an HTTP request
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+
+public class HttpRequest {
```

Comment:
@pluraliseseverythings The decision of adding our own classes such as HttpRequest, and HttpResponse was a deliberate one as I want to decouple the interface from implementation. With this approach, changes in the implementation won't leak into the interface.

Do you prefer if we use Jersey?

Reply:
Good point. Maybe let's make it a pure interface there and let's call it GraknEngineEndpoint so it doesn't mention Http.
I don't have a 100% clear understanding of this but the red flag for me is something like HttpResponse. Initially it's just a status code and body but then we have to build all the seralisation into body string, make a method of the type isSuccessful() and all the other stuff that is already implemented in other libraries.

Reply:
@pluraliseseverythings Honestly, I'm a bit worried about that as well. Is Jersey considered "standard" or "implementation neutral"? If so we can switch to that instead.

Reply:
JAX-RS is the standard implemented in Jersey. Most of the Jersey classes are JAX-RS

Reply:
@pluraliseseverythings The two HTTP classes you linked above are actually not `public`.

Reply:
Sorry, I'm trying to give feedback without having much context here. The right class to reference is probably this 

https://github.com/jax-rs/api/blob/master/jaxrs-api/src/main/java/javax/ws/rs/core/Request.java

Although I'm not sure it does/abstracts what we need.  The other problem is, until we migrate to JAX-RS (I'm on it now) it will be a pain to integrate with Spark. 

I'd say maybe let's keep these classes you made but make sure we don't implement too much http filtering logic. That comes off the shelf (e.g. https://jersey.github.io/documentation/latest/client.html#d0e5221)

Reply:
@pluraliseseverythings cool. Let's do just that and we can revisit this point afterwards

Reply:
I will track it here: https://work.grakn.ai/entity/17646